lib/ansible/modules/cloud/vmware/vmware_guest_customization.py

```diff
@@ -0,0 +1,566 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_customization
+short_description: Manage virtual machine customization specifications
+description:
+    - This module can be used to create, delete, update and rename virtual machine's customization specifications.
+version_added: 2.9
+author:
+- Abhijeet Kasurde (@Akasurde)
+notes:
+    - Tested on ESXi 6.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+    spec_name:
+      description:
+      - Unique name identifying the customization specification.
+      - This parameter is case sensitive.
+      - This name can be used as C(customization_spec) is M(vmware_guest) module for customizing operating systems.
+      required: True
+      type: str
+    spec_type:
+      description:
+      - Type of network identity and settings.
+      - If set to C(Linux), then LinuxPrep is used for the customization. Used for Linux like operating systems.
+      - If set to C(Windows), then SysPrep is used for the customization. Used for Windows like operating systems.
+      - If not set, then C(Linux) is used as default.
+      default: 'Linux'
+      choices: ['Linux', 'Windows']
+      type: str
+    spec_description:
+      description:
+      - The description for the given customization specification.
+      type: str
+    networks:
+      description:
+      - A list of networks.
+      - This is a required parameter, if C(state) is set to C(present).
+      - All parameters and VMware object names are case sensitive.
+      - 'One of the below parameters is required per entry:'
+      - ' - C(type) (string): Type of IP address assignment (either C(dhcp) or C(static)). C(dhcp) is default.'
+      - ' - C(ip) (string): Static IP address and implies C(type) as C(static).'
+      - ' - C(netmask) (string): Static netmask required for C(ip) and implies C(type) as C(static).'
+      - ' - C(gateway) (string): Static gateway.'
+      - ' - C(dns_servers) (string): DNS servers for this network interface (Windows).'
+      - ' - C(domain) (string): Domain name for this network interface (Windows).'
+      type: list
+    dns_servers:
+      description:
+      - A list of DNS servers.
+      - This parameter applies to both Linux and Windows customization.
+      type: list
+    dns_suffix:
+      description:
+      - 'A list of domain suffixes, also known as DNS search path (default: C(domain) parameter).'
+      - This parameter applies to both Linux and Windows customization.
+      type: list
+    domain:
+      description:
+      - DNS domain name to use.
+      - This parameter applies to both Linux and Windows customization.
+      type: str
+    vm_hostname:
+      description:
+      - Short Full Qualified Domain Name.
```

Comment:
```suggestion
      - Short Name.
```
Or possibly just 'Host name for the VM'.